pycbc/inference/models/relbin.py

```diff
@@ -0,0 +1,355 @@
+        dictionary. The inner products in each detector will be
+        weighted by 1/psd of that detector.
+    mass1 : float
+        The primary mass in solar masses used for generating the fiducial
+        waveform.
+    mass2 : float
+        The secondary mass in solar masses used for generating the fiducial
+        waveform.
+    spin1z : float
+        The component of primary dimensionless spin along the orbital angular
+        momentum used for generating the fiducial waveform.
+    spin2z : float
+        The component of secondary dimensionless spin along the orbital angular
+        momentum used for generating the fiducial waveform.
+    ra : float
+        The right ascension in radians used for generating the fiducial
+        waveform.
+    dec : float
+        The declination in radians used for generating the fiducial waveform.
+    tc : float
+        The GPS time of coalescence used for generating the fiducial waveform.
+    low_frequency_cutoff : float
+        The starting frequency used in computing inner products. This will be
+        the same for all detectors.
+    high_frequency_cutoff : float
+        The ending frequency used in computing inner products. This will be
+        the same for all detectors. Note that no checks are made to ensure
+        that a template waveform does not end below this frequency, so
+        choose this value to be lower than any expected final frequency
+        for template waveforms.
+    epsilon : float, optional
+        Tuning parameter used in calculating the frequency bins. Lower values
+        will result in higher resolution and more bins.
+    \**kwargs :
+        All other keyword arguments are passed to ``BaseDataModel``.
+    """
+    name = "relative"
+
+    def __init__(self, data, psds, mass1, mass2, spin1z, spin2z,
+                 ra, dec, tc, low_frequency_cutoff,
+                 high_frequency_cutoff, epsilon=0.5, **kwargs):
+        super(Relative, self).__init__(data=data, **kwargs)
+        # store data and frequencies
+        f_lo = float(low_frequency_cutoff)
+        f_hi = float(high_frequency_cutoff)
+        self.f = numpy.array(data[data.keys()[0]].sample_frequencies)
+        self.df = data[data.keys()[0]].delta_f
+        self.data = data
+        self.end_time = float(data[data.keys()[0]].end_time)
+        self.det = {ifo: Detector(ifo) for ifo in data}
+        epsilon = float(epsilon)
+        # store data and psds as arrays for faster computation
+        self.comp_data = {ifo: numpy.array(data[ifo].data) for ifo in data}
+        self.comp_psds = {ifo: numpy.array(psds[ifo].data) for ifo in data}
+        self._psds = psds
+        # store fiducial waveform params
+        mass1 = float(mass1)
+        mass2 = float(mass2)
+        spin1z = float(spin1z)
+        spin2z = float(spin2z)
+        ra = float(ra)
+        dec = float(dec)
+        tc = float(tc)
+
+        # get detector-specific arrival times relative to end of data
```

Comment:
One change you've made from my earlier implementation is that this goes back to including time in as a parameter which needs to be also modeled in the frequency shift. I see now why you wanted to provide a sky location as you need to determine the time of the peak likelihood in each detector. If this is far off, the approximation could get problematic. Is this right? How much is it a problem then if you don't have the right sky location or don't have the reference time to more than 50-100 ms ?

Reply:
In principle yes, setting fiducial parameters far from the peak will cause problems for the approximation, but in practice the method seems fairly robust to inaccuracies in sky location and coalescence time. For example, changing the above config settings for GW170817 to
```
ra = 2.0
dec = 0.0
tc = 1187008882.5
```
produces very similar posteriors. I don't have a more general answer yet for how far is "too far" but I'll look at mapping that out in the near future.